Documentation/devel/distribution.md

```diff
@@ -0,0 +1,147 @@
+
+**NOTE:** I'm not sure if distribution is the best name for this concept or it should be called with other names like "image reference". So I'd like to hear your thoughts.
```

Comment:
I guess we can settle on calling them "Distribution Points". Current name looks fine enough, we just need to disambiguate it from typical "Linux distro" usage.